backend/services/user_service.py

```
@@ -112,6 +113,89 @@ async def create_user(db: AsyncSession, user_data: UserCreate) -> User:
+        
+        # Check if email already exists (ultra-optimized EXISTS query)
+        # This prevents duplicate accounts and maintains data integrity
+        email_start = time.perf_counter()
+        email_exists_query = select(exists().where(User.email == user_data.email))
+        email_check = await db.execute(email_exists_query)
+        email_exists = email_check.scalar()
+        email_time = time.perf_counter() - email_start
+        logger.info(f"📧 Email check completed in: {email_time:.3f}s")
+        
+        if email_exists:
+            raise ValueError("Email already registered")
```

Ruff (0.13.2)

Avoid specifying long messages outside the exception class

(TRY003)